### PR TITLE
SapMachine #1544: Run wiki update task regularly

### LINKED ISSUE #1544: Run wiki update task regularly
Update [Blogs](https://github.com/SAP/SapMachine/wiki/Blogs) wiki page regularly using a GitHub action.

### PR DESCRIPTION
**Pull request by parttimenerd**
fixes #1544

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @parttimenerd, this pull request fulfills all formal requirements.

File: .github/actions/update-wiki/action.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Runs update-wiki update actions every day at 20:00 UTC
+
+name: Update Wiki
+
+on:
+  push:
+    branches:
+      - master
```
Comment (RealCLanger) [inline .github/actions/update-wiki/action.yml:8]: I think the on section is not needed here as the update-wiki action is triggered by the wiki workflow.

File: .github/workflows/wiki.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: 'Wiki-related'
+
+on:
+  push:
```
Comment (RealCLanger) [inline .github/workflows/wiki.yml:4]: No reason for on push here, the cron schedule is the thing. However, you could add a "workflow-dispatch" trigger to enable updating manually

**Review by RealCLanger — CHANGES_REQUESTED** [review]

File: .github/actions/update-wiki/action.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  schedule:
+    - cron: '0 20 * * *'
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v4
```
Comment (RealCLanger) [inline .github/actions/update-wiki/action.yml:15]: Why is the checkout here necessary? You already do it in the workflow...

File: .github/actions/update-wiki/action.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    - name: Update Wiki
+      run: |
+          git clone https://github.com/SAP/SapMachine.wiki.git
+          cd SapMachine.wiki
+          pip3 install feedparser
+          python3 scripts/update_blogs.py update > ../out
```
[the remark this answers is not in the document]
Comment (parttimenerd) [reply .github/actions/update-wiki/action.yml:21]: it's in the wiki repo: https://github.com/SAP/SapMachine.wiki.git

[the remark this answers is not in the document]
Comment (parttimenerd) [reply .github/actions/update-wiki/action.yml:21]: I didn't want to move it too far away from the wiki sources.

[the remark this answers is not in the document]
Comment (RealCLanger) [reply .github/actions/update-wiki/action.yml:16]: Good. 😄

File: .github/actions/update-wiki/action.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Runs update-wiki update actions every day at 20:00 UTC
```
Comment (RealCLanger) [inline .github/actions/update-wiki/action.yml:1]: This comment is more appropriate in .github/workflows/wiki.yml

File: .github/actions/update-wiki/action.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+name: Update Wiki
+
+description: |
+  This workflow updates the wiki with the latest blog posts.
+
```
Comment (RealCLanger) [inline .github/actions/update-wiki/action.yml:7]: Should be: "This action ..."

File: .github/actions/update-wiki/action.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+          git clone https://github.com/SAP/SapMachine.wiki.git
+          cd SapMachine.wiki
+          pip3 install feedparser
+          python3 scripts/update_blogs.py update > ../out
+          cat ../out
+          if grep -q changed ../out; then
+              git commit -a -m "Update blog posts"
+              git push
+              echo "Blog post list updated"
+          fi
```
Comment (RealCLanger) [inline .github/actions/update-wiki/action.yml:22]: Please also add an output when no updates occured, like:
...
else
    echo "No updates"
...

File: .github/actions/update-wiki/action.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      working-directory: .
+      shell: bash
```
Comment (RealCLanger) [inline .github/actions/update-wiki/action.yml:24]: Add line break at the end.

File: .github/workflows/wiki.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+name: 'Wiki-related'
```
Comment (RealCLanger) [inline .github/workflows/wiki.yml:1]: This is the name we see in the Github Actions. So it should be a bit nicer, like 'Wiki Update'.

File: .github/workflows/wiki.yml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+on:
+  workflow_dispatch: {}
```
Comment (RealCLanger) [inline .github/workflows/wiki.yml:4]: is the `: {}` needed here?

**Review by RealCLanger — CHANGES_REQUESTED** [review]
Looks good overall. Some final polishing requested.

File: .github/workflows/wiki.yml (hunk 10 above)
Comment (parttimenerd) [reply .github/workflows/wiki.yml:4]: I don't know, YAML is weird

**Review by RealCLanger — APPROVED** [review]

**Comment by SapMachine** [comment]
Hello @parttimenerd, this pull request fulfills all formal requirements.

**Review by MBaesken — APPROVED** [review]

**Pull request merged**